pydantic_ai_slim/pydantic_ai/_otel_messages.py

```diff
@@ -25,36 +31,57 @@ class ToolCallPart(TypedDict):
 
 
 class ToolCallResponsePart(TypedDict):
+    """A tool call response part in a GenAI message."""
+
     type: Literal['tool_call_response']
     id: str
     name: str
+    # TODO: This should be `response` not `result`
     result: NotRequired[JsonValue]
     builtin: NotRequired[bool]  # Not (currently?) part of the spec, used by Logfire
 
 
-class MediaUrlPart(TypedDict):
-    type: Literal['image-url', 'audio-url', 'video-url', 'document-url']
-    url: NotRequired[str]
+class UriPart(TypedDict):
+    """A URI part in a GenAI message (for images, audio, video, documents).
 
+    Per the semantic conventions, uses 'uri' type with modality field.
+    """
 
-class BinaryDataPart(TypedDict):
-    type: Literal['binary']
-    media_type: str
-    content: NotRequired[str]
+    type: Literal['uri']
+    uri: NotRequired[str]
+    modality: NotRequired[str]
+
+
+class BlobPart(TypedDict):
+    """A blob (binary data) part in a GenAI message.
 
+    Per the semantic conventions, uses 'blob' type with modality field.
+    """
 
-class ThinkingPart(TypedDict):
-    type: Literal['thinking']
+    type: Literal['blob']
+    blob: NotRequired[str]
+    modality: NotRequired[str]
+
+
+class ReasoningPart(TypedDict):
+    """A reasoning/thinking part in a GenAI message.
+
+    Per the semantic conventions, uses 'reasoning' type.
```

Comment:
You can find `UriPart`, `BlobPart`, and `ReasoningPart` in https://github.com/open-telemetry/semantic-conventions/blob/main/docs/gen-ai/gen-ai-input-messages.json with this schema. Not sure how practical it is for us to change this in the immediate future but I think we'll want to comply with those conventions asap..